### PR TITLE
Throw an exception if a non-SHA1 digest is passed

### PR DESCRIPTION
**Pull request by whikloj**
Addresses [FCREPO-2075](https://jira.duraspace.org/browse/FCREPO-2075)

### CODE REVIEW

File: fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java [hunk 1]
```diff
@@ -786,16 +788,24 @@ private static void checkLinkForLdpResourceCreation(final String link) {
...
         try {
             final Map<String,String> digestPairs = RFC3230_SPLITTER.split(nullToEmpty(digest));
-            return digestPairs.entrySet().stream()
-                .filter(s -> s.getKey().toLowerCase().equals("sha1"))
-                .map(Map.Entry::getValue)
-                .findFirst()
-                .map("urn:sha1:"::concat)
-                .orElse("");
+            final Set<String> checksumTypes =
+                    digestPairs.keySet().stream().map(s -> s.toLowerCase()).collect(Collectors.toSet());
```
Comment (ajs6f) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java]: `s -> s.toLowerCase()` is `String::toLowerCase`.

File: fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java [hunk 2]
```diff
@@ -786,16 +788,24 @@ private static void checkLinkForLdpResourceCreation(final String link) {
...
+            // If you have one or more digests and one is sha1 or no digests.
+            if (digestPairs.isEmpty() || checksumTypes.contains("sha1")) {
```
Comment (ajs6f) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java]: If you don't need to do anything else with `checksumTypes`, then you could do:
```
boolean checksumTypeIncludeSHA = digestPairs.keySet().stream().map(String::toLowerCase).anyMatch("sha1");
```

Comment (whikloj) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java]: Thank you, I knew there was something cleaner. Just couldn't pull it together.

File: fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java [hunk 3]
```diff
@@ -793,10 +791,10 @@ private static void checkLinkForLdpResourceCreation(final String link) {
     private static String parseDigestHeader(final String digest) throws InvalidChecksumException {
         try {
             final Map<String,String> digestPairs = RFC3230_SPLITTER.split(nullToEmpty(digest));
-            final Set<String> checksumTypes =
-                    digestPairs.keySet().stream().map(s -> s.toLowerCase()).collect(Collectors.toSet());
+            final boolean checksumTypeIncludeSHA1 = digestPairs.keySet().stream().map(String::toLowerCase)
+                .anyMatch("sha1"::equals);
```
Comment (ajs6f) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java]: Ah, I'm a dope-- I totally forgot `String::equalsIgnoreCase`, so just:
```
digestPairs.keySet().stream().anyMatch("sha1"::equalsIgnoreCase)
```

Comment (whikloj) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java]: Oh yeah, I kept looking for case insensitive matching in the `Set<String>` when I started this. Now that we are back to just using `String` I never thought about those useful methods. 👍

**Comment by awoods** [comment]
:+1: Looks good to me. @ajs6f, feel free to squash/merge when you are content.

**Pull request merged**